File: ManualTransition/Week2 Selenium Rebuild Properly/Practical/selenium-page-object-model-practice/src/main/java/com/techlambda/selenium/pom/pages/DashboardPage.java
package com.techlambda.selenium.pom.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * DashboardPage represents dashboard-related locators and actions.
 *
 * This class hides dashboard UI details from the test class.
 */
public class DashboardPage {

    private final WebDriver driver;

    private final By dashboardStatus = By.id("dashboard-status");
    private final By welcomeMessage = By.id("welcome-message");
    private final By searchBox = By.id("search-box");
    private final By searchButton = By.id("search-button");
    private final By searchResult = By.id("search-result");

    public DashboardPage(WebDriver driver) {
        this.driver = driver;
    }

    public String getDashboardStatus() {
        return driver.findElement(dashboardStatus).getText();
    }

    public String getWelcomeMessage() {
        return driver.findElement(welcomeMessage).getText();
    }

    public void searchProduct(String productName) {
        driver.findElement(searchBox).clear();
        driver.findElement(searchBox).sendKeys(productName);
        driver.findElement(searchButton).click();
    }

    public String getSearchResult() {
        return driver.findElement(searchResult).getText();
    }
}

File: PlayWriter/Practical/Advance/playwright-reusable-architecture-practice/pages/DashboardPage.ts
import { Locator, Page } from '@playwright/test';

// PAGE LAYER / POM
// DashboardPage contains locators and actions related to dashboard only.
export class DashboardPage {
  readonly page: Page;
  readonly title: Locator;
  readonly welcomeMessage: Locator;
  readonly profileRole: Locator;
  readonly logoutButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.title = page.getByTestId('dashboard-title');
    this.welcomeMessage = page.getByTestId('welcome-message');
    this.profileRole = page.getByTestId('profile-role');
    this.logoutButton = page.getByTestId('logout-button');
  }

  async logout(): Promise<void> {
    await this.logoutButton.click();
  }
}
